dropshot_endpoint/src/endpoint.rs

```diff
@@ -1082,7 +1082,14 @@ mod tests {
         );
     }
 
-    // These argument types are close to being invalid, but are actually valid.
+    // These argument types are close to being invalid, but are either okay or
+    // we're not sure.
+    //
+    // * `MyRequestContext` might be a type alias, which is legal for
+    //   function-based servers.
```

Comment:
will type aliases not valid for trait-based servers?

Reply:
Yeah, for trait-based servers, type aliases can't be used for `RequestContext` specifically. We need to be able to replace the type within for our purposes.